qa/rpc-tests/p2p-segwit.py

```diff
@@ -985,6 +998,14 @@ def test_block_relay(self, segwit_activated):
             assert_equal(wit_block.serialize(False), non_wit_block.serialize())
             assert_equal(wit_block.serialize(True), block.serialize(True))
 
+            # Upgraded node should not ask for blocks from unupgraded
+            block4 = self.build_next_block(nVersion=4)
+            block4.solve()
+            self.old_node.getdataset = set()
+            # Wait for tx to arrive, check if last_last_getdata is nonempty
+            self.old_node.announce_block(block4, use_header=False)
+            self.old_node.announce_tx_and_wait_for_getdata(block4.vtx[0])
```

Comment:
Comment needs to be updated.  Perhaps also explain the ~~hack~~ trick used here?  Eg:

```
# Blocks can be requested via direct-fetch (immediately upon processing the announcement) 
# or via parallel download (with an indeterminate delay from processing the announcement), so 
# to test that a block is NOT requested, we could guess a time period to sleep for, and then check.  
# We can avoid the sleep() by taking advantage of transaction getdata's being processed after block
# getdata's, and announce a transaction as well, and then check the first getdata message received.
```